src/frontend/src/icp/components/receive/IcReceive.svelte

```diff
@@ -34,13 +34,15 @@
 	$: icrc = token.standard === 'icrc';
 
 	const open: LoadTokenAndOpenModal = async (callback: () => Promise<void>) => {
-		tokenStore.set(token);
-		await callback();
+		await loadTokenAndRun({ token, callback });
 	};
 
+	const dispatch = createEventDispatcher();
+
 	const close: CloseModalAndResetToken = () => {
 		modalStore.close();
-		tokenStore.set(null);
+		// We are resetting the token in the parent. That way we can know if we are using a global page $token or a selected token.
+		dispatch('nnsClose');
```

Comment:
if we are thinking of abandoning the page `$token` at some point, maybe it is worth to start pointing the places where will we modify once done? some TODO here maybe? or we revisit the entire code once done?

Reply:
Not sure, ultimatelly I kind of feel like `$token` should just not exist at all anymore, so it would be like adding TODO everywhere ;)